testing/src/main/java/io/stargate/it/http/graphql/graphqlfirst/FederationTest.java

```diff
@@ -131,4 +132,25 @@ public void udtKeyTest() {
 
     assertThat(JsonPath.<Integer>read(response, "$._entities[0].k.k")).isEqualTo(1);
   }
+
+  @Test
+  @DisplayName("Should include trace if requested via header")
+  public void federatedTracing() {
+    Object response =
+        CLIENT.getKeyspaceFullResponse(
+            ImmutableMap.of("apollo-federation-include-trace", "ftv1"),
+            KEYSPACE,
+            "query {\n"
+                + "_entities(representations: [ "
+                + "{ __typename: \"Entity2\", k: 1 }, "
+                + " ]) { "
+                + "... on Entity2 { k } "
+                + "} }");
+
+    String trace = JsonPath.read(response, "$.extensions.ftv1");
+    // The value is an opaque string (Base64 encoding of the protobuf representation of the trace).
```

Comment:
interesting, could you elaborate more on how this `trace_id` will be used by end-users?

Reply:
The trace data is consumed by the [Apollo Gateway](https://www.apollographql.com/docs/federation/metrics/):

> These federated traces capture the subgraph-level details in the shape of the query plan, which is sent to Apollo's metrics ingress by default, and aggregated into query-level stats and field-level stats.